options/options.go

```diff
@@ -325,7 +325,13 @@ func (interp *interpreter) interpretFieldOptions(fqn string, fld *descriptorpb.F
 	if index >= 0 {
 		opt := uo[index]
 		optNode := interp.file.OptionNode(opt)
-		if extendee := fld.GetExtendee(); extendee != "" && fld.GetJsonName() != string(opt.GetStringValue()) {
+		var jsonName string
+		if opt.StringValue != nil {
+			jsonName = string(opt.StringValue)
+		}
```

Comment:
Out of curiosity, why this, instead of simply moving the `if opt.StringValue == nil` check below up here?

In the case that a file has something like `json_name = 123`, I'm trying to decide if it is more confusing to say "option json_name is not allowed on extensions" instead of "expecting string value for json_name option". I think I might expect the type error first, and expect the other error if the value is actually a string.

Reply:
Updated.